### ISSUE TITLE
Gradle Windows scripts don't propagate exit codes correctly

### ISSUE THREAD

**Issue by mataha** [gradle/gradle#15428]
Gradle scripts on Windows don't care what exit code an underlying app has returned - as long as it's non-zero, they will return `1`. This behaviour is inconsistent with how these scripts work on Unix-based operating systems.

### Expected Behavior
An exact exit code is returned.

### Current Behavior
`0` on success, `1` on anything else (regardless of an underlying exit code). Both `gradlew.bat` and scripts generated by `installDist` task behave like this.

### Context
I was writing command-line scripts to complement the ones created by `installDist` in order to pass command-line arguments more naturally (i.e. not by `gradlew run --args='--something --verbose -u'`) when I've noticed this behaviour. Ideally I would like to capture exact exit codes out-of-the-box.

### Steps to Reproduce 
I've illustrated an example here: https://github.com/mataha/gradle-windows-exit-code

```batchfile
git clone https://github.com/mataha/gradle-windows-exit-code
cd gradle-windows-exit-code
:: Expectedly should print: `Exit code: 2`
run 2
```

Output: `Exit code: 1`

**Comment by ljacomet** [comment]
Fixed through #19865

**Comment by ljacomet** [comment]
Thank you for the report _and_ the fix!
